src/Aspire.Dashboard/Components/Dialogs/SettingsDialog.razor.cs

```diff
@@ -36,27 +32,21 @@ protected override void OnInitialized()
         });
     }
 
-    protected override async Task OnAfterRenderAsync(bool firstRender)
+    private async Task SettingChangedAsync()
     {
-        if (firstRender)
+        // The field is being transiently set to null when the value changes. Maybe a bug in FluentUI?
+        // This should never be set to null by the dashboard so we can ignore null values.
+        if (_currentSetting != null)
         {
-            _jsModule = await JS.InvokeAsync<IJSObjectReference>("import", "/js/app-theme.js");
-            _currentSetting = await _jsModule.InvokeAsync<string>("getThemeCookieValue");
-            StateHasChanged();
+            // The theme isn't changed here. Instead, the MainLayout subscribes to the change event
+            // and applies the new theme to the browser window.
+            await ThemeManager.RaiseThemeChangedAsync(_currentSetting);
         }
     }
 
-    private async Task SettingChangedAsync(string newValue)
-    {
-        // The theme isn't changed here. Instead, the MainLayout subscribes to the change event
-        // and applies the new theme to the browser window.
-        _currentSetting = newValue;
-        await ThemeManager.RaiseThemeChangedAsync(newValue);
-    }
-
-    public async ValueTask DisposeAsync()
+    public ValueTask DisposeAsync()
```

Comment:
Can we implement IDisposable instead of IAsyncDisposable if we're not awaiting anything here?